examples/show_auto_updated_partition_manager.py

```diff
@@ -0,0 +1,177 @@
+used; the longer durations indicate that a list operation was performed on the
+HMC.
+"""
+
+import sys
+import uuid
+# import logging
+import time
+import requests.packages.urllib3
+from time import sleep
+from datetime import datetime, timedelta
+
+import zhmcclient
+from zhmcclient.testutils import hmc_definitions
+
+requests.packages.urllib3.disable_warnings()
+
+# Get HMC info from HMC inventory and vault files
+hmc_def = hmc_definitions()[0]
+nickname = hmc_def.nickname
+host = hmc_def.host
+userid = hmc_def.userid
+password = hmc_def.password
+verify_cert = hmc_def.verify_cert
+
+# logging.basicConfig(level=logging.DEBUG, stream=sys.stderr)
+
+def delta_ms(start_dt, end_dt):
+    """Return float that is the time delta in milliseconds"""
+    return (end_dt - start_dt) / timedelta(microseconds=1) / 1000
+
+print(__doc__)
+
+print("Using HMC {} at {} with userid {} ...".format(nickname, host, userid))
+
+print("Creating a session with the HMC ...")
+try:
+    session = zhmcclient.Session(
+        host, userid, password, verify_cert=verify_cert)
+except zhmcclient.Error as exc:
+    print("Error: Cannot establish session with HMC {}: {}: {}".
+          format(host, exc.__class__.__name__, exc))
+    sys.exit(1)
+
+cleanup_partition = None
+try:
+    client = zhmcclient.Client(session)
+
+    print("Finding a CPC in DPM mode ...")
+    cpcs = client.cpcs.list(filter_args={'dpm-enabled': True})
+    if not cpcs:
+        print("Error: HMC at {} does not manage any CPCs in DPM mode".
+              format(host))
+        sys.exit(1)
+    cpc = cpcs[0]
+    print("Using CPC {}".format(cpc.name))
+
+    part_mgr = cpc.partitions
+
+    print("Listing partitions using list() (not auto-enabled) ...")
+    start_dt = datetime.now()
+    part_list = part_mgr.list()
+    end_dt = datetime.now()
+    print("Result of list(): returned {} partitions in {} ms".
+          format(len(part_list), delta_ms(start_dt, end_dt)))
+
+    print("Enabling auto-updating for partition manager")
+    try:
+        part_mgr.enable_auto_update()
+    except zhmcclient.Error as exc:
+        print("Error: Cannot auto-enable partition manager for CPC {}: {}: {}".
+              format(cpc.name, exc.__class__.__name__, exc))
+        sys.exit(1)
+
+    print("Listing partitions using list() "
+          "(auto-enabled - uses from local list) ...")
+    start_dt = datetime.now()
+    part_list = part_mgr.list()
+    end_dt = datetime.now()
+    print("Result of list(): returned {} partitions in {} ms".
+          format(len(part_list), delta_ms(start_dt, end_dt)))
+
+    part_name = "zhmc_test_{}".format(uuid.uuid4())
+    print("Creating partition {} ...".format(part_name))
+    part_props = {
+        'name': part_name,
+        'description': 'Original partition description.',
+        'cp-processors': 2,
+        'initial-memory': 1024,
+        'maximum-memory': 2048,
+        'processor-mode': 'shared',
+        'boot-device': 'test-operating-system'
```

Comment:
We should not use 'test-operating-system' - it is not documented and not available on all machines.

Reply:
This was just a rename of the script, but ok so let's fix it. I have removed the boot-device property as it is not needed to the purpose of showing property updates.